source/develop/release-management/features/network/managed_mtu_for_vm_networks.html.md

```diff
@@ -0,0 +1,332 @@
+the [largest possible MTU size that can safely be applied][5] to VMs,
+but in oVirt there will be no automatic calculation of the MTU.
+
+### MTU individually per network
+MTU has to be managed individually for each network, because:
+* The MTU of the logical network are transported either as plain Ethernet
+  frames or encapsulated in a VLAN is individual for each network.
+* The MTU for the external logical networks depends on the MTU of the underlying
+  network and for localnets on the MTU of the connected logical network, too.
+  There might be no common underlying network for the tunnels and this way it
+  the common MTU is not obvious.
+An individual MTU per network requires that oVirt engine supports the MTU
+property of external networks:
+* On Importing a network created outside engine on a provider.
+* During automatic synchronization of external networks.
+This requires enhancements in ovirt-engine, [OpenStack Java SDK][11] and
+ovirt-provider-ovn.
+
+### Propagating the MTU into the VMs
+
+#### Set the MTU by DHCP
+The MTU can be propagated via DHCP into the VMs. Since oVirt implements L3
+functionality only for external logical network, this approach works only for
+external logical networks and if a subnet is defined for the logical network
+and the VM configures the network interface via DHCP.
+
+The only DHCP server managed by oVirt is the OVN internal DHCP server, which
+is utilized to implement OpenStack API's subnets. Currently the DHCP server is
+able to propagate a common MTU for all networks, configured in the
+configuration file of ovirt-provider-ovn.
+If we would [support the property MTU][8] of external networks, we could
+propagate the MTU from ovirt-engine via the ovirt-provider-ovn into OVN's DHCP
+server. This would require enhancements in ovirt-engine, [OpenStack Java
+SDK][11] and ovirt-provider-ovn.
+An alternative approach is to disable setting the MTU via DHCP in default
+configuration.
+
+DHCP can only be used to reduce the MTU inside the guest to values smaller than
+1500 bytes, because DHCP does not configure the tap devices of the VM.
+
+#### Set the MTU by libvirt
+Since Red Hat Enterprise Linux 7.4 it is possible to set the MTU in libvirt's
+XML, which will be propagated to libvirt's tap devices, qemu and virtio drivers
+inside the guest VM.
+
+The MTU is respected by QEMU for rhel7.4.0 machine type and later.
+It is ignored by QEMU with rhel7.3.0 machine type and earlier because the MTU
+feature wasn't supported, so migration from a RHEL 7.4 host to a RHEL 7.3 host
+would fail.
+
+TODO: In which cluster level rhel7.4.0 machine type will be default?
+
+If an oVirt VM with rhel7.4.0 machine type connected to ovirtmgmt with a MTU of
+1234 and an OVN network with a MTU of 1100 is running, the configuration on the
+host looks like this:
+
+~~~~
+[root@host23 ~]# ip l
+35: br-int: <BROADCAST,MULTICAST> mtu 1100 qdisc noop state DOWN mode DEFAULT qlen 1000
```

Comment:
mtu 1100 - this might suggest that this value would be set by libvirt too.

Reply:
yes, it is propagated.